app/src/features/surveys/observations/sampling-sites/edit/SamplingSiteEditPage.tsx

```diff
@@ -0,0 +1,144 @@
+import Box from '@mui/material/Box';
+import CircularProgress from '@mui/material/CircularProgress';
+import { grey } from '@mui/material/colors';
+import { IErrorDialogProps } from 'components/dialog/ErrorDialog';
+import { CreateSamplingSiteI18N } from 'constants/i18n';
+import { DialogContext } from 'contexts/dialogContext';
+import { SurveyContext } from 'contexts/surveyContext';
+import { FormikProps } from 'formik';
+import History from 'history';
+import { APIError } from 'hooks/api/useAxios';
+import { useBiohubApi } from 'hooks/useBioHubApi';
+import { useContext, useRef, useState } from 'react';
+import { Prompt, useHistory, useParams } from 'react-router';
+import SamplingSiteHeader from '../SamplingSiteHeader';
+import SampleSiteEditForm, { IEditSamplingSiteRequest } from './components/SampleSiteEditForm';
+
+const SamplingSiteEditPage = () => {
+  const history = useHistory();
+  const biohubApi = useBiohubApi();
+  const urlParams: Record<string, string | number | undefined> = useParams();
+  const surveySampleSiteId = Number(urlParams['survey_sample_site_id']);
+
+  const surveyContext = useContext(SurveyContext);
+  const dialogContext = useContext(DialogContext);
+
+  const formikRef = useRef<FormikProps<IEditSamplingSiteRequest>>(null);
+
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [enableCancelCheck, setEnableCancelCheck] = useState(true);
+
+  const showCreateErrorDialog = (textDialogProps?: Partial<IErrorDialogProps>) => {
+    dialogContext.setErrorDialog({
+      dialogTitle: CreateSamplingSiteI18N.createErrorTitle,
+      dialogText: CreateSamplingSiteI18N.createErrorText,
+      onClose: () => {
+        dialogContext.setErrorDialog({ open: false });
+      },
+      onOk: () => {
+        dialogContext.setErrorDialog({ open: false });
+      },
+      ...textDialogProps,
+      open: true
+    });
+  };
+
+  const handleSubmit = async (values: IEditSamplingSiteRequest) => {
+    try {
+      setIsSubmitting(true);
+
+      await biohubApi.samplingSite.editSampleSite(
+        surveyContext.projectId,
+        surveyContext.surveyId,
+        surveySampleSiteId,
+        values
+      );
+
+      // Disable cancel prompt so we can navigate away from the page after saving
+      setEnableCancelCheck(false);
+
+      // Refresh the context, so the next page loads with the latest data
+      surveyContext.sampleSiteDataLoader.refresh(surveyContext.projectId, surveyContext.surveyId);
+
+      // create complete, navigate back to observations page
+      history.push(`/admin/projects/${surveyContext.projectId}/surveys/${surveyContext.surveyId}/observations`);
+    } catch (error) {
+      showCreateErrorDialog({
+        dialogTitle: CreateSamplingSiteI18N.createErrorTitle,
+        dialogText: CreateSamplingSiteI18N.createErrorText,
+        dialogError: (error as APIError).message,
+        dialogErrorDetails: (error as APIError)?.errors
+      });
+      setIsSubmitting(false);
```

Comment:
Discussed: since this submit function navigates the user away after submitting, we don't need to disable the loading spinner (also we can't call setState after we navigate, since the component is being destroyed already).